de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/bpartner/model/interceptor/C_BPartner.java

```diff
@@ -60,4 +70,28 @@ public void createBPartnerStatsRecord(@NonNull final I_C_BPartner bpartner)
 	{
 		Services.get(IBPartnerStatsDAO.class).retrieveBPartnerStats(bpartner);
 	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_CHANGE}, ifColumnsChanged = {I_C_BPartner.COLUMNNAME_C_BP_Group_ID})
+	public void updateBPartnerStatsRecord(@NonNull final I_C_BPartner bpartner)
+	{
+		final IBPartnerStats bpartnerStats = Services.get(IBPartnerStatsDAO.class).retrieveBPartnerStats(bpartner);
+		final I_C_BP_Group bpGroup = bpartner.getC_BP_Group();
+		final String creditStatus = bpGroup.getSOCreditStatus();
+		if (!Check.isEmpty(creditStatus,true))
+		{
+			setSO_CreditStatus(bpartnerStats, creditStatus);
```

Comment:
also i am a bit puzzled... wondering, isn't the job of the updater to set the SO_CreditStatus, and mainly, to update the C_BPartner_Stats?

Reply:
yes; But, at first, when we create the partner, we need the creditstatus based on bpartner group.